apps/artemis-web/utils/captchaMemoryStore.ts

```diff
@@ -0,0 +1,38 @@
+function MemoryStore(windowMs) {
+    let hits = {};
+
+    this.decr = function (key) {
+        if (hits[key] && hits[key] !== 0) {
+            hits[key]--;
+        }
+
+        return hits[key];
+    };
+
+    this.incr = function (key) {
+        if (hits[key]) {
+            hits[key]++;
```

Comment:
this should happen only if `hits[key] < CAPTCHA_TRIES`. Otherwise we may increase the counter to inifinity, which we do not need to. If `hits` reached the max limit, then it can stay there; the captcha will be activated until the counter is reset or decreased anyway (reset seems a better solution for captcha since it is an on-off feature; if you login OK, then reset either via LDAP or locally, otherwise increase until limit is hit)

Reply:
fixed :)